cmd/catchpointdump/file.go

```diff
@@ -0,0 +1,225 @@
+			}
+		}
+		err = catchupAccessor.ProgressStagingBalances(ctx, header.Name, balancesBlockBytes, &downloadProgress)
+		if err != nil {
+			return fileHeader, err
+		}
+		if header.Name == "content.msgpack" {
+			// we already know it's valid, since we validated that above.
+			protocol.Decode(balancesBlockBytes, &fileHeader)
+		}
+	}
+}
+
+func printAccountsDatabase(databaseName string, fileHeader ledger.CatchpointFileHeader, outFile *os.File) error {
+	dbAccessor, err := db.MakeAccessor(databaseName, true, false)
+	if err != nil || dbAccessor.Handle == nil {
+		return err
+	}
+	if fileHeader.Version != 0 {
+		fmt.Fprintf(outFile, "Version: %d\nBalances Round: %d\nBlock Round: %d\nBlock Header Digest: %s\nCatchpoint: %s\nTotal Accounts: %d\nTotal Chunks: %d\n",
+			fileHeader.Version,
+			fileHeader.BalancesRound,
+			fileHeader.BlocksRound,
+			fileHeader.BlockHeaderDigest.String(),
+			fileHeader.Catchpoint,
+			fileHeader.TotalAccounts,
+			fileHeader.TotalChunks)
+
+		totals := fileHeader.Totals
+		fmt.Fprintf(outFile, "AccountTotals - Online Money: %d\nAccountTotals - Online RewardUnits : %d\nAccountTotals - Offline Money: %d\nAccountTotals - Offline RewardUnits : %d\nAccountTotals - Not Participating Money: %d\nAccountTotals - Not Participating Money RewardUnits: %d\nAccountTotals - Rewards Level: %d\n",
+			totals.Online.Money.Raw, totals.Online.RewardUnits,
+			totals.Offline.Money.Raw, totals.Offline.RewardUnits,
+			totals.NotParticipating.Money.Raw, totals.NotParticipating.RewardUnits,
+			totals.RewardsLevel)
+	}
+	return dbAccessor.Atomic(func(tx *sql.Tx) (err error) {
+		if fileHeader.Version == 0 {
+			var totals ledger.AccountTotals
+			id := ""
+			row := tx.QueryRow("SELECT online, onlinerewardunits, offline, offlinerewardunits, notparticipating, notparticipatingrewardunits, rewardslevel FROM accounttotals WHERE id=?", id)
+			err = row.Scan(&totals.Online.Money.Raw, &totals.Online.RewardUnits,
+				&totals.Offline.Money.Raw, &totals.Offline.RewardUnits,
+				&totals.NotParticipating.Money.Raw, &totals.NotParticipating.RewardUnits,
+				&totals.RewardsLevel)
+			if err != nil {
+				return err
+			}
+			fmt.Fprintf(outFile, "AccountTotals - Online Money: %d\nAccountTotals - Online RewardUnits : %d\nAccountTotals - Offline Money: %d\nAccountTotals - Offline RewardUnits : %d\nAccountTotals - Not Participating Money: %d\nAccountTotals - Not Participating Money RewardUnits: %d\nAccountTotals - Rewards Level: %d\n",
+				totals.Online.Money.Raw, totals.Online.RewardUnits,
+				totals.Offline.Money.Raw, totals.Offline.RewardUnits,
+				totals.NotParticipating.Money.Raw, totals.NotParticipating.RewardUnits,
+				totals.RewardsLevel)
+		}
+
+		balancesTable := "accountbase"
+		if fileHeader.Version != 0 {
+			balancesTable = "catchpointbalances"
+		}
+		rows, err := tx.Query(fmt.Sprintf("SELECT address, data FROM %s order by address", balancesTable))
+		if err != nil {
+			return
+		}
+		defer rows.Close()
+
+		for rows.Next() {
```

Comment:
Printing in json format prints too much data: slow, redundant, and difficult to read. 
Why not print a header row, and put only the values in the following rows? 
Is every jsonData of the same structure?

Reply:
no, the jsonData might not be the same for all the rows.
I know that there is a lot more to be desired here, but keep in mind that it was developed as a debugging tool - we wanted to compare two catchpoints and review differences, if any. If it will become an ongoing used tool, we should definitely consider additional format options.